b64ct/src/errors.rs

```diff
@@ -0,0 +1,63 @@
+use core::fmt;
+
+/// Insufficient output buffer length.
+#[derive(Copy, Clone, Debug, Eq, PartialEq)]
+pub struct InvalidLengthError;
+
+impl fmt::Display for InvalidLengthError {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> Result<(), fmt::Error> {
+        f.write_str("insufficient output buffer length")
+    }
+}
+
+#[cfg(feature = "std")]
+impl std::error::Error for InvalidLengthError {}
+
+/// Invalid encoding of provided "B64" string.
+#[derive(Copy, Clone, Debug, Eq, PartialEq)]
+pub struct InvalidEncodingError;
+
+impl fmt::Display for InvalidEncodingError {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> Result<(), fmt::Error> {
+        f.write_str("insufficient output buffer length")
+    }
+}
+
+#[cfg(feature = "std")]
+impl std::error::Error for InvalidEncodingError {}
+
+/// Generic error, union of [`InvalidLengthError`] and [`InvalidEncodingError`].
+#[derive(Copy, Clone, Debug, Eq, PartialEq)]
+pub enum Error {
+    /// Insufficient output buffer length.
+    InvalidEncoding,
+    /// Invalid encoding of provided "B64" string.
+    InvalidLength,
+}
+
+impl fmt::Display for Error {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> Result<(), fmt::Error> {
+        match self {
+            v @ Self::InvalidEncoding => v.fmt(f),
+            v @ Self::InvalidLength => v.fmt(f),
```

Comment:
Looks like this is breaking the build due to infinite recursion. Would suggest something like:

```suggestion
            v @ Self::InvalidEncoding => f.write_str("invalid B64 encoding"),
            v @ Self::InvalidLength => f.write_str("invalid B64 length"),
```

Reply:
Hm, weird... It was building fine on my PC.